src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -2154,6 +2154,96 @@ void MacroAssembler::check_klass_subtype_slow_path(Register sub_klass,
   else if (result_reg == noreg) { blr(); } // return with CR0.eq if neither label nor result reg provided
 
   bind(fallthru);
+  if (L_success != nullptr && result_reg == noreg) {
```

Comment:
Is there a problem if `L_success == nullptr && result_reg == noreg` and there aren't any secondary supers?
In that case we would reach here with CR0.eq from L2134 and we would fallthrough with CR0.eq. Due to the change in `C1StubId::slow_subtype_check_id` we would return there with CR0.eq.

Reply:
This is a reproducer:
```java
public class InstanceOfTest {

    public static interface TestInterfaceI {
    }

    public static class TestClassNegative {
    }

    public static void main(String[] args) {
        Object obj = new TestClassNegative();
        for (int i = 100_000; i > 0; i--) {
            dontinline_testMethod(obj);
        }
        boolean result = dontinline_testMethod(obj);
        System.out.println("result: " + result);
    }

    static boolean dontinline_testMethod(Object obj) {
        return obj instanceof TestInterfaceI;
    }
}
```
```
./jdk/bin/java -XX:TieredStopAtLevel=1 -XX:-UseSecondarySupersTable InstanceOfTest
result: true
```